q2cli/core/usage.py

```diff
@@ -74,14 +75,18 @@ def to_interface_name(self):
 
         return cli_name
 
-    def assert_has_line_matching(self, path, expression):
+    def assert_has_line_matching(self, path, expression, key=None):
         if not self.use.enable_assertions:
             return
 
         INDENT = self.use.INDENT
         input_path = self.to_interface_name()
         expr = shlex.quote(expression)
 
+        if key:
+            input_path = \
+                "%s%s%s" % (input_path, key, self.execute()[key].extension)
```

Comment:
Yeah, it seemed like overkill to us too - but the Usage API has no way of knowing what the output type is until the factory has been executed. I think we'd have to add some new functionality that would allow for that, but not sure if the juice is worth the squeeze. We'll see what we can do this afternoon and circle back if there isn't a promising solution.